zetaclient/zetacore_observer.go

```diff
@@ -123,66 +123,57 @@ func (co *CoreObserver) startSendScheduler() {
+							params := send.GetCurrentOutTxParam()
+							if params.ReceiverChainId != c.ChainId {
+								log.Warn().Msgf("mismatch chainid: want %d, got %d", c.ChainId, params.ReceiverChainId)
 								continue
 							}
 
 							// Monitor Core Logger for OutboundTxTssNonce
-							included, _, err := ob.IsSendOutTxProcessed(send.Index, int(send.GetCurrentOutTxParam().OutboundTxTssNonce), send.GetCurrentOutTxParam().CoinType, co.logger.ZetaChainWatcher)
+							included, _, err := ob.IsSendOutTxProcessed(send.Index, params.OutboundTxTssNonce, params.CoinType, co.logger.ZetaChainWatcher)
 							if err != nil {
-								co.logger.ZetaChainWatcher.Error().Err(err).Msgf("IsSendOutTxProcessed fail %s", c.ChainName)
+								co.logger.ZetaChainWatcher.Error().Err(err).Msgf("IsSendOutTxProcessed fail, Chain ID: %s", c.ChainName)
 								continue
 							}
 							if included {
 								co.logger.ZetaChainWatcher.Info().Msgf("send outTx already included; do not schedule")
 								continue
 							}
-							chain, err := GetTargetChain(send)
-							if err != nil {
-								co.logger.ZetaChainWatcher.Error().Err(err).Msgf("GetTargetChain fail , Chain ID : %s", c.ChainName)
-								continue
-							}
-							nonce := send.GetCurrentOutTxParam().OutboundTxTssNonce
-							outTxID := fmt.Sprintf("%s-%d-%d", send.Index, send.GetCurrentOutTxParam().ReceiverChainId, nonce) // should be the outTxID?
+							nonce := params.OutboundTxTssNonce
+							outTxID := fmt.Sprintf("%s-%d-%d", send.Index, params.ReceiverChainId, nonce) // should be the outTxID?
 
 							// FIXME: config this schedule; this value is for localnet fast testing
 							if bn >= math.MaxInt64 {
 								continue
 							}
 							currentHeight := uint64(bn)
-							var interval uint64
-							var lookahead int64
-							// FIXME: fix these ugly type switches and conversions
-							switch v := ob.(type) {
-							case *EVMChainClient:
-								interval = uint64(v.GetChainConfig().CoreParams.OutboundTxScheduleInterval)
-								lookahead = v.GetChainConfig().CoreParams.OutboundTxScheduleLookahead
-							case *BitcoinChainClient:
-								interval = uint64(v.GetChainConfig().CoreParams.OutboundTxScheduleInterval)
-								lookahead = v.GetChainConfig().CoreParams.OutboundTxScheduleLookahead
-							default:
-								co.logger.ZetaChainWatcher.Error().Msgf("unknown ob type on chain %s: type %T", chain, ob)
-								continue
-							}
+							interval := uint64(ob.GetCoreParameters().OutboundTxScheduleInterval)
+							lookahead := uint64(ob.GetCoreParameters().OutboundTxScheduleLookahead)
 							if nonce%interval == currentHeight%interval && !outTxMan.IsOutTxActive(outTxID) {
 								outTxMan.StartTryProcess(outTxID)
-								co.logger.ZetaChainWatcher.Debug().Msgf("chain %s: Sign outtx %s with value %d\n", chain, send.Index, send.GetCurrentOutTxParam().Amount)
+								co.logger.ZetaChainWatcher.Debug().Msgf("chain %s: Sign outtx %s with value %d sats", chain, send.Index, params.Amount)
```

Comment:
```suggestion
								co.logger.ZetaChainWatcher.Debug().Msgf("chain %s: Sign outtx %v with value %d sats", chain, send.Index, params.Amount)
```
Index is not a string isn't it?

Reply:
Thought on this?